angular/test/protractor/group_spec.coffee

```diff
@@ -12,6 +12,26 @@ describe 'Group Page', ->
         page.expectElement('link[rel=prev]')
         page.expectElement('link[rel=next]')
 
+      it 'should allow you to join an open group', ->
+        page.loadPath 'view_open_group_as_visitor'
+        page.click '.join-group-button__join-group'
+        browser.driver.findElement(By.id('user_name')).sendKeys('Name')
+        browser.driver.findElement(By.id('user_email')).sendKeys('test@example.com')
+        browser.driver.findElement(By.id('user_password')).sendKeys('complex_password')
+        browser.driver.findElement(By.id('user_password_confirmation')).sendKeys('complex_password')
+        browser.driver.findElement(By.id('create-account')).click()
+        page.click '.group-welcome-modal__close-button'
+        page.expectElement '.lmo-navbar__item--user'
+        page.expectElement '.group-theme__name', 'Open Dirty Dancing Shoes'
+
+      it 'should allow you to request to join a closed group', ->
```

Comment:
If I'm joining a closed group:

- Stay on the group page and give me a flash message saying I've requested membership
- Once I get an invitation, that flow should work the same in angular as it did with beta.